internal/service/certificatesigningrequest.go

```diff
@@ -244,6 +241,13 @@ func (h *ServiceHandler) PatchCertificateSigningRequest(ctx context.Context, req
 func (h *ServiceHandler) ReplaceCertificateSigningRequest(ctx context.Context, request server.ReplaceCertificateSigningRequestRequestObject) (server.ReplaceCertificateSigningRequestResponseObject, error) {
 	orgId := store.NullOrgId
 
+	// check that the CSR does not already exist
+	_, err := h.store.CertificateSigningRequest().Get(ctx, orgId, request.Name)
+	if !errors.Is(err, flterrors.ErrResourceNotFound) {
+		msg := fmt.Sprintf("could not create CSR resource: CSR with name %s already exists", request.Name)
+		return server.ReplaceCertificateSigningRequest409JSONResponse{Message: msg}, nil
+	}
+
```

Comment:
A PUT verb on a resource has the behaviour "CreateOrUpdate", so it shouldn't be an error if the CSR already exists?

Reply:
There are problems with auto-approving the CSR twice, so I will need to revisit this to see if there is another alternative besides throwing the error.

Reply:
Update: I am going to make these changes in a separate PR.